Day1/DayOnePartOne.py

```diff
@@ -0,0 +1,72 @@
+class FloorCounter:
+    
+    current_floor = 0
+    instructions = None
+    parsed_instructions = []
+    going_up = None
+
+    def __init__(self, instructions, up_char="(", down_char=")"):
+        self.instructions = instructions
+        self.up_char = up_char
+        self.down_char = down_char
+        self.parsed_instructions = []
+        self.going_up = None
+
+    def parse_character(self, character):
+        if character == self.up_char:
+            self.parsed_instructions.append(1)
+        elif character == self.down_char:
+            self.parsed_instructions.append(-1)
+        elif character == "+":
+            self.going_up = True
+        elif character == "-":
+            self.going_up = False
+        elif character.isnumeric():
+            character = int(character)
+            if self.going_up:
+                self.parsed_instructions.append(character)
+            else:
+                self.parsed_instructions.append(-abs(character))
+        else:
+            raise ValueError(f"Input character not recognized: {character}")
+
+    def parse_input(self):
+        for character in self.instructions:
+            self.parse_character(character)
+        return self.parsed_instructions
+
+    def is_next_floor_skipped(self, skipped_floors, elevator_up=True):
+        if elevator_up:
+            return str(self.current_floor+1) in skipped_floors
+        else:
+            return str(self.current_floor-1) in skipped_floors
+    
+    def change_floors(self, move):
+        self.current_floor += move
+
+    def change_floors_with_skip(self, move, skipped_floors):
```

Comment:
```
def change_floors_with_skip(self, move, skipped_floors):
    increment = 1 if move > 0 else -1
    steps = abs(move)
    for _ in range(steps):
        self.current_floor += increment
        while self.current_floor in skipped_floors:
            self.current_floor += increment
```
Ha tudsz akkor probald meg osszevonni a scenariokat, nem kezeled igazan mashogy a move < 0-t sem, csak nem 1-et adsz hozza hanem -1-et.
L56-ban sem az erdekel hogy mi a tenyleges erteke az i-nek (nem is hasznalod) hanem hogy hanyszor loop-olsz, akkor nem kell negativ szamokon loop-olnod es maris ugyanaz a ket scenario

Reply:
Absztrakciokban gondolkodj. Mi a +1, mi a -1? tudod-e valami kozos absztrakcioval jelolni (increment). Min loop-olok itt, szamit hogy mi az i? vagy csak hogy hanyat lepek adott iranyba (steps).